platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java

```diff
@@ -0,0 +1,364 @@
+        if (certProps == null) {
+            errors.add("No cert properties sent.");
+            return errors;
+        }
+        String keystoreFileName = (String) certProps.get(KEYSTORE_FILE_NAME);
+        String keystoreData = (String) certProps.get(KEYSTORE_FILE);
+        String keystorePass = (String) certProps.get(KEYSTORE_PASS);
+        String keyPass = (String) certProps.get(KEY_PASS);
+        String truststoreFileName = (String) certProps.get(TRUSTSTORE_FILE_NAME);
+        String truststoreData = (String) certProps.get(TRUSTSTORE_FILE);
+        String truststorePass = (String) certProps.get(TRUSTSTORE_PASS);
+        Boolean devMode = ((Boolean) certProps.get(DEV_MODE));
+
+        String fqdn = baseUrl.getHost();
+
+        if (devMode != null && devMode) {
+            try {
+                setupDevCerts(fqdn, oldHostName);
+            } catch (MalformedObjectNameException e) {
+                LOGGER.error("Failed to create dev certs.", e);
+                errors.add("Failed to create dev certificates. " + e.getMessage());
+            }
+            return errors;
+        }
+
+        if (StringUtils.isEmpty(keystoreFileName) || StringUtils.isEmpty(keystoreData)
+                || StringUtils.isEmpty(keystorePass) || StringUtils.isEmpty(keyPass)) {
+            errors.add("Some of the required keystore fields are missing");
+        }
+
+        if (StringUtils.isEmpty(truststoreFileName) || StringUtils.isEmpty(truststoreData)
+                || StringUtils.isEmpty(truststorePass)) {
+            errors.add("Some of the required truststore fields are missing");
+        }
+
+        if (errors.size() > 0) {
+            return errors;
+        }
+
+        try {
+            KeystoreEditorMBean keystoreMbeanProxy = getKeystorEditorMbean();
+
+            if (keystoreMbeanProxy.keystoreContainsEntry(fqdn, keystorePass, keystoreData, null,
+                    keystoreFileName)) {
+
+                keystoreMbeanProxy.addAllKeystoreEntries(keyPass, keystorePass, keystoreData, null,
+                        keystoreFileName);
+                keystoreMbeanProxy
+                        .addAllTruststoreEntries(null, truststorePass, truststoreData, null,
+                                truststoreFileName);
+
+                //remove old private key
+                if (!oldHostName.equals(fqdn)) {
+                    keystoreMbeanProxy.deletePrivateKey(oldHostName);
+                }
+            } else {
+                errors.add("Keystore does not contain the required key for " + fqdn);
+            }
+        } catch (KeystoreEditor.KeystoreEditorException e) {
+            errors.add("Error adding system key to keystore. " + e.getMessage());
+            LOGGER.error("Error adding system key to keystore.", e);
+        } catch (Exception e) {
+            errors.add("Error communicating with KeystoreEditorMBean. " + e.getMessage());
+            LOGGER.error("JMX error updating certs", e);
```

Comment:
Why aren't the errors and logger messages the same here, although they are above? 
Should they be the same or is there a reason why they're different here?